lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -656,10 +713,21 @@ public long ramBytesUsed() {
     // threads when IndexSearcher is created with threads
     private final AtomicBoolean used;
 
+    private final Executor executor;
+
     CachingWrapperWeight(Weight in, QueryCachingPolicy policy) {
       super(in.getQuery(), 1f);
       this.in = in;
       this.policy = policy;
+      this.executor = null;
+      used = new AtomicBoolean(false);
+    }
+
+    CachingWrapperWeight(Weight in, QueryCachingPolicy policy, Executor executor) {
+      super(in.getQuery(), 1f);
+      this.in = in;
+      this.policy = policy;
+      this.executor = executor;
       used = new AtomicBoolean(false);
     }
```

Comment:
nit: can we have a single constructor? The first one looks unused?

Reply:
Good catch, thanks.